src/data_source/storage/sql.rs

```diff
@@ -672,38 +793,71 @@ impl VersionedDataSource for SqlStorage {
 // These tests run the `postgres` Docker image, which doesn't work on Windows.
 #[cfg(all(any(test, feature = "testing"), not(target_os = "windows")))]
 pub mod testing {
+    #![allow(unused_imports)]
+    use crate::data_source::storage::sql::sqlx::ConnectOptions;
     use async_compatibility_layer::art::async_timeout;
     use async_std::net::TcpStream;
+    use rand::Rng;
+    use refinery::Migration;
+    use sqlx::sqlite::SqliteConnectOptions;
     use std::{
-        env,
+        env, fs,
+        path::Path,
         process::{Command, Stdio},
-        str,
+        str::{self, FromStr},
         time::Duration,
     };
 
     use portpicker::pick_unused_port;
-    use refinery::Migration;
 
-    use super::Config;
+    use super::{Config, Db};
     use crate::testing::sleep;
 
     #[derive(Debug)]
     pub struct TmpDb {
+        #[cfg(not(feature = "embedded-db"))]
         host: String,
+        #[cfg(not(feature = "embedded-db"))]
         port: u16,
+        #[cfg(not(feature = "embedded-db"))]
         container_id: String,
+        #[cfg(feature = "embedded-db")]
+        db_path: std::path::PathBuf,
+        #[cfg(not(feature = "embedded-db"))]
         persistent: bool,
```

Comment:
We could implement this behavior for sqlite too. It would just mean not deleting the file on drop. This is nice sometimes for debugging because the DB is still there after the test fails, so you can look into it

Reply:
yeah, I think there is also a bug with Drop implementation for posgres.